pimd/pim_iface.c

```diff
@@ -401,19 +394,18 @@ static int pim_sec_addr_update(struct interface *ifp)
 	}
 
 	for (ALL_LIST_ELEMENTS_RO(ifp->connected, node, ifc)) {
-		struct prefix *p = ifc->address;
+		pim_addr addr = pim_addr_from_prefix(ifc->address);
 
-		if (PIM_INADDR_IS_ANY(p->u.prefix4)) {
+		if (pim_addr_is_any(addr))
 			continue;
-		}
 
-		if (pim_ifp->primary_address.s_addr == p->u.prefix4.s_addr) {
+		if (pim_addr_cmp(addr, pim_ifp->primary_address)) {
```

Comment:
"!" check missing here for prim-addr equal comparision & skip

Reply:
All other changes LGTM.

Reply:
Good catch! Fixed.